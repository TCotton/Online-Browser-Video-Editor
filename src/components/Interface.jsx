import React from "react";
import {Button} from "reakit/Button";
import {Tooltip, TooltipReference, useTooltipState} from "reakit/Tooltip";
import FileUpload from './PlayerComponents/InputFile';
import ErrorBoundary from './ErrorBoundary/Error';
import {Play} from './PlayerDisplay/Play';
import {Pause} from './PlayerDisplay/Pause';
import {Forward} from './PlayerDisplay/Forward';
import {Backward} from './PlayerDisplay/Backward';
import {Sound} from './PlayerDisplay/Sound';
import {FileExport} from "./FileExportComponents/FileExport";
import {AutoLoad} from "./AutoLoad/AutoLoad";

export default function Interface() {
    const tooltip3 = useTooltipState({baseId: 'tooltip3'});
    const tooltip4 = useTooltipState({baseId: 'tooltip4'});
    const tooltip5 = useTooltipState({baseId: 'tooltip5'});
    const tooltip6 = useTooltipState({baseId: 'tooltip6'});
    const tooltip7 = useTooltipState({baseId: 'tooltip7'});
    const tooltip8 = useTooltipState({baseId: 'tooltip8'});
    const tooltip9 = useTooltipState({baseId: 'tooltip9'});

    return (
        <ErrorBoundary>
            <ul>
                <li>
                    <button>
                        <img src={"./cut.svg"} className="right-cut" alt="right cut"/>
                    </button>
                </li>
                <li>
                    <TooltipReference {...tooltip3} as={Button}>
                        <Backward/>
                    </TooltipReference>
                    <Tooltip {...tooltip3}>Backward</Tooltip>
                </li>
                <li>
                    <TooltipReference {...tooltip8} as={Button}>
                        <Pause/>
                    </TooltipReference>
                    <Tooltip {...tooltip8}>Stop</Tooltip>
                </li>
                <li>
                    <TooltipReference {...tooltip4} as={Button}>
                        <Play/>
                    </TooltipReference>
                    <Tooltip {...tooltip4}>Play</Tooltip>
                </li>
                <li>
                    <TooltipReference {...tooltip5} as={Button}>
                        <Forward/>
                    </TooltipReference>
                    <Tooltip {...tooltip5}>Forward</Tooltip>
                </li>

                <li>
                    <TooltipReference {...tooltip9} as={Button}>
                        <Sound/>
                    </TooltipReference>
                    <Tooltip {...tooltip9}>Mute/Sound</Tooltip>
                </li>

                <li>
                    <TooltipReference {...tooltip6}>
                        <FileUpload/>
                    </TooltipReference>
                    <Tooltip {...tooltip6}>Upload</Tooltip>
                </li>
                <li>
                    <TooltipReference {...tooltip7}>
                        <ErrorBoundary>
                            <FileExport/>
                        </ErrorBoundary>
                    </TooltipReference>
                    <Tooltip {...tooltip7}>Export</Tooltip>
                </li>
            </ul>
        </ErrorBoundary>
    )
}