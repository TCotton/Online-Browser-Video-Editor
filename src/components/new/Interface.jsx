import React from "react";
import {Button} from "reakit/Button";
import {Tooltip, TooltipReference, useTooltipState} from "reakit/Tooltip";

import FileUpload from './InputFile';
import ErrorBoundary from '../ErrorBoundary/Error';
import {fileName} from './slices/filesSlice';

export default function Interface() {
    const tooltip1 = useTooltipState({baseId: 'tooltip1'});
    const tooltip2 = useTooltipState({baseId: 'tooltip2'});
    const tooltip3 = useTooltipState({baseId: 'tooltip3'});
    const tooltip4 = useTooltipState({baseId: 'tooltip4'});
    const tooltip5 = useTooltipState({baseId: 'tooltip5'});
    const tooltip6 = useTooltipState({baseId: 'tooltip6'});
    const tooltip7 = useTooltipState({baseId: 'tooltip7'});

    return (
        <ErrorBoundary>
            <ul>
                <li>
                    File: {fileName}
                </li>
                <li>
                    <TooltipReference {...tooltip1} as={Button}>
                        <img src={"./cut.svg"} className="left-cut" alt="left cut"/>
                    </TooltipReference>
                    <Tooltip {...tooltip1}>Drap and drop to use as first edit point</Tooltip>
                </li>
                <li>
                    <TooltipReference {...tooltip2} as={Button}>
                        <img src={"./cut.svg"} className="right-cut" alt="right cut"/>
                    </TooltipReference>
                    <Tooltip {...tooltip2}>Drap and drop to use as the second edit point</Tooltip>
                </li>
                <li>
                    <TooltipReference {...tooltip3} as={Button}>
                        <img src={"./left-frame.svg"} className="left-frame" alt="left frame"/>
                    </TooltipReference>
                    <Tooltip {...tooltip3}>Left frame</Tooltip>
                </li>
                <li>
                    <TooltipReference {...tooltip4} as={Button}>
                        <img src={"./play.svg"} className="play" alt="play"/>
                    </TooltipReference>
                    <Tooltip {...tooltip4}>Play</Tooltip>
                </li>
                <li>
                    <TooltipReference {...tooltip5} as={Button}>
                        <img src={"./right-frame.svg"} className="right-frame" alt="right frame"/>
                    </TooltipReference>
                    <Tooltip {...tooltip5}>Right Frame</Tooltip>
                </li>
                <li>
                    <TooltipReference {...tooltip6}>
                        <FileUpload/>
                    </TooltipReference>
                    <Tooltip {...tooltip6}>Upload</Tooltip>
                </li>
                <li>
                    <TooltipReference {...tooltip7} as={Button}>
                        <img src={"./export.svg"} className="export" alt="export"/>
                    </TooltipReference>
                    <Tooltip {...tooltip7}>Export</Tooltip>
                </li>
            </ul>
        </ErrorBoundary>
    )
}