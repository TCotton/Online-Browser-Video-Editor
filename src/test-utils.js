import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { createStore, combineReducers } from 'redux'
import { Provider } from 'react-redux'

import counterReducer from './components/slices/counterSlice';
import filesReducer from './components/slices/filesSlice';
import videoReducer from './components/slices/videoSlice';
import playerReducer from './components/slices/playerSlice';
import audioReducer from './components/slices/audioSlice';
import imageReducer from './components/slices/imageSlice';
import waveformReducer from './components/slices/waveformSlice';

function render(
    ui,
    {
        initialState,
        store = createStore(combineReducers({
            counter: counterReducer,
            files:  filesReducer,
            video: videoReducer,
            player: playerReducer,
            audio: audioReducer,
            images: imageReducer,
            waveform: waveformReducer,
        }), initialState),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react'
// override render method
export { render }