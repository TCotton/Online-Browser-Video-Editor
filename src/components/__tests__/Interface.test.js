import React from 'react'
import Interface from '../Interface'
import { render } from '../../test-utils'

jest.mock('reakit/Tooltip', () => ({
  ...jest.requireActual('reakit/Tooltip'),
  useTooltipState: jest.fn(n => n)
}))

describe('Interface', () => {
  it('should render successfully', () => {
    const { container } = render(<Interface />)
    expect(container.querySelector('[data-testid="interface"]')).not.toBeNull()
  })

  it('should match snapshot', () => {
    const { baseElement } = render(<Interface />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should render exact aria-describedby as provided by the useTooltipState hook', () => {
    const { container } = render(<Interface />)
    expect(
      container.querySelector("[aria-describedby='tooltip3']")
    ).not.toBeNull()
    expect(
      container.querySelector("[aria-describedby='tooltip4']")
    ).not.toBeNull()
    expect(
      container.querySelector("[aria-describedby='tooltip5']")
    ).not.toBeNull()
    expect(
      container.querySelector("[aria-describedby='tooltip6']")
    ).not.toBeNull()
    expect(
      container.querySelector("[aria-describedby='tooltip7']")
    ).not.toBeNull()
    expect(
      container.querySelector("[aria-describedby='tooltip8']")
    ).not.toBeNull()
    expect(
      container.querySelector("[aria-describedby='tooltip9']")
    ).not.toBeNull()
  })
})
