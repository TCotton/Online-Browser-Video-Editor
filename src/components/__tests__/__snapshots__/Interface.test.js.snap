// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interface should match snapshot 1`] = `
<body>
  <div>
    <ul
      data-testid="interface"
    >
      <li>
        <button>
          <img
            alt="right cut"
            class="right-cut"
            src="./cut.svg"
          />
        </button>
      </li>
      <li>
        <button
          aria-describedby="tooltip3"
          tabindex="0"
          type="button"
        >
          <img
            alt="backward"
            class="backward"
            data-testid="reverse"
            src="./back.svg"
          />
        </button>
      </li>
      <li>
        <button
          aria-describedby="tooltip8"
          tabindex="0"
          type="button"
        >
          <img
            alt="stop"
            class="stop"
            data-testid="pause"
            src="./pause.svg"
          />
        </button>
      </li>
      <li>
        <button
          aria-describedby="tooltip4"
          tabindex="0"
          type="button"
        >
          <img
            alt="play"
            class="play"
            data-testid="play"
            src="./play.svg"
          />
        </button>
      </li>
      <li>
        <button
          aria-describedby="tooltip5"
          tabindex="0"
          type="button"
        >
          <img
            alt="forward"
            class="forward"
            data-testid="forward"
            src="./forward.svg"
          />
        </button>
      </li>
      <li>
        <button
          aria-describedby="tooltip9"
          tabindex="0"
          type="button"
        >
          <img
            alt="sound"
            class="sound"
            data-testid="sound"
            src="./sound.svg"
          />
        </button>
      </li>
      <li>
        <div
          aria-describedby="tooltip6"
          tabindex="0"
        >
          <div
            class="app"
          >
            <form
              action="#"
              data-testid="form"
            >
              <div
                class="fileStyling"
              >
                <input
                  accept="video/*"
                  data-testid="file"
                  id="file"
                  name="file"
                  required=""
                  type="file"
                  value=""
                />
                <label
                  for="file"
                >
                  Choose a file
                </label>
              </div>
              <div
                class="Errors"
              />
            </form>
          </div>
        </div>
      </li>
      <li>
        <div
          aria-describedby="tooltip7"
          tabindex="0"
        >
          <button
            aria-controls="FileExport"
            aria-expanded="false"
            aria-haspopup="dialog"
            type="button"
          >
            <img
              alt="export"
              class="export"
              data-testid="file-export"
              src="./export.svg"
            />
          </button>
        </div>
      </li>
    </ul>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      hidden=""
      id="tooltip3"
      role="tooltip"
      style="display: none; pointer-events: none;"
    >
      Backward
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      hidden=""
      id="tooltip8"
      role="tooltip"
      style="display: none; pointer-events: none;"
    >
      Stop
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      hidden=""
      id="tooltip4"
      role="tooltip"
      style="display: none; pointer-events: none;"
    >
      Play
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      hidden=""
      id="tooltip5"
      role="tooltip"
      style="display: none; pointer-events: none;"
    >
      Forward
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      hidden=""
      id="tooltip9"
      role="tooltip"
      style="display: none; pointer-events: none;"
    >
      Mute/Sound
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      hidden=""
      id="tooltip6"
      role="tooltip"
      style="display: none; pointer-events: none;"
    >
      Upload
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      class="css-jhczzu"
      data-dialog-ref="FileExport"
      hidden=""
      style="display: none;"
    >
      <div
        aria-label="Welcome"
        aria-modal="true"
        class="css-jkbxbt"
        data-dialog="true"
        hidden=""
        id="FileExport"
        role="dialog"
        style="display: none;"
        tabindex="-1"
      >
        <div>
          <div
            class="export"
            data-testid="export"
          >
            <header>
              <p
                class="title"
              >
                Click below to export
              </p>
              <p
                class="status"
              />
            </header>
            <ul
              class="social"
            >
              <li>
                <button
                  data-testid="twitter"
                >
                  <img
                    alt="click to export for twitter"
                    height="400"
                    src="./social/twitter.svg"
                    width="400"
                  />
                </button>
              </li>
              <li>
                <button
                  data-testid="facebook"
                >
                  <img
                    alt="click to export for Facebook"
                    height="250"
                    src="./social/facebook.png"
                    width="250"
                  />
                </button>
              </li>
              <li>
                <button
                  data-testid="linkedin"
                >
                  <img
                    alt="click to export for Linkedin"
                    height="540"
                    src="./social/linkedin.png"
                    width="650"
                  />
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="__reakit-portal"
  >
    <div
      hidden=""
      id="tooltip7"
      role="tooltip"
      style="display: none; pointer-events: none;"
    >
      Export
    </div>
  </div>
</body>
`;
